.github/workflows/ci.yml

```diff
@@ -38,39 +52,35 @@ jobs:
         python -m pip install --upgrade pip
         pip install protobuf==3.7.1
 
-    # Cache yarn deps. thx https://github.com/actions/cache/blob/master/examples.md#node---yarn
-    - name: Get yarn cache directory path
-      id: yarn-cache-dir-path
-      run: echo "::set-output name=dir::$(yarn cache dir)"
-    - name: Set up node_modules cache
-      uses: actions/cache@v1
-      id: yarn-cache
-      with:
-        path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
-        key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
-        restore-keys: |
-          ${{ runner.os }}-yarn-
-
     - run: yarn --frozen-lockfile
     - run: yarn build-all
     - run: yarn diff:sample-json
-    - run: yarn type-check
-    - run: yarn lint
     - run: yarn test-proto # Run before unit-core because the roundtrip json is needed for proto tests.
 
     # Run tests that require headfull Chrome.
     - run: sudo apt-get install xvfb
-    - run: xvfb-run --auto-servernum yarn unit
-    - run: xvfb-run --auto-servernum yarn test-clients
-    - run: xvfb-run --auto-servernum yarn smoke --debug -j=1 --retries=2
-    - run: xvfb-run --auto-servernum yarn test-bundle
-    - run: xvfb-run --auto-servernum yarn test-docs
+    - name: yarn unit
+      run: xvfb-run --auto-servernum yarn unit
+    - name: yarn test-clients
+      run: xvfb-run --auto-servernum yarn test-clients
+    - name: yarn test-bundle
+      run: xvfb-run --auto-servernum yarn test-bundle
+    - name: yarn test-docs
+      run: xvfb-run --auto-servernum yarn test-docs
 
     - run: yarn test-lantern
     - run: yarn test-legacy-javascript
```

Comment:
Some of these tests _can_ take a long time. That's why I split up the "special" tests (legacy js, lantern, etc.) from the core unit test and the build test–with this PR all of these are still in the same job.